iot_config/src/org.rs

```diff
@@ -138,7 +151,179 @@ pub async fn create_org(
     Ok(org)
 }
 
-async fn insert_constraints(
+pub async fn update_org(
+    oui: u64,
+    authorizer: UpdateAuthorizer,
+    updates: Vec<proto::UpdateV1>,
+    db: impl sqlx::PgExecutor<'_> + sqlx::Acquire<'_, Database = sqlx::Postgres>,
+) -> Result<Org, OrgStoreError> {
+    let mut txn = db.begin().await?;
+
+    let net_id = get_org_netid(oui, &mut txn).await?;
+    let is_helium_org = [
+        HeliumNetId::Type0.id(),
+        HeliumNetId::Type3.id(),
+        HeliumNetId::Type6.id(),
+    ]
+    .contains(&net_id);
+    let current_org = get(oui, &mut txn).await?;
+
+    for update in updates {
+        match update.update {
+            Some(proto::Update::Owner(pubkeybin)) if authorizer == UpdateAuthorizer::Admin => {
+                update_owner(oui, pubkeybin.into(), &mut txn).await?
+            }
+            Some(proto::Update::Devaddrs(addr_count))
+                if authorizer == UpdateAuthorizer::Admin && is_helium_org =>
+            {
+                let helium_net_id: HeliumNetId = net_id
+                    .try_into()
+                    .map_err(|err: &'static str| OrgStoreError::InvalidUpdate(err.to_string()))?;
+                let constraints = helium_netids::checkout_devaddr_constraints(
+                    &mut txn,
+                    addr_count,
+                    helium_net_id,
+                )
+                .await
+                .map_err(|err| OrgStoreError::SaveConstraints(format!("{err}")))?;
+                insert_helium_constraints(oui, net_id, &constraints, &mut txn).await?
+            }
+            Some(proto::Update::Constraint(constraint_update))
+                if authorizer == UpdateAuthorizer::Admin && is_helium_org =>
```

Comment:
This check prevents roaming orgs from allowing others access to their devaddr ranges.

Reply:
beyond the initial creation rpc that allocates the devaddr contraints/netid we don't currently distinguish between roamers and helium orgs and the current requirement was that orgs could modify their own delegate keys and payer key while a (foundation) admin must authorize a change of the owner pubkey and adding/removing devaddr constraints

Reply:
additionally, i don't disagree with letting roaming orgs manage the devaddr constraints within their owned non-helium netids or even helium orgs subletting portions of their own leased constraint ranges but since this wasn't an immediately necessary feature i'd rather not support it until we have time to do a pretty substantial reworking of how to do authorization and permissioning of access to rpcs and records; i can't really conceive of a way to allow those kinds of distinctions without reworking the entire auth structure and probably a fair portion of the org/route database table schemas

Reply:
I was only trying to point out the `&& is_helium_org` prevents any non-helium org from being updated by this rpc. Whether that be a roaming org subletting part of their address space, or a typo on creation. 

But it's not something needed right now. Just mentioning it in hopes that someone will remember when reworking is done. :+1: